frontend/src/metabase-lib/order_by.unit.spec.ts

```diff
@@ -0,0 +1,75 @@
+import type { Field, TableId } from "metabase-types/api";
+import { createQuery, SAMPLE_DATABASE, SAMPLE_METADATA } from "./test-helpers";
+import * as ML from "./v2";
+
+function getColumnCount(tableId: TableId) {
+  return SAMPLE_METADATA.table(tableId)?.fields.length || 0;
+}
+
+describe("order by", () => {
+  describe("orderableColumns", () => {
+    const query = createQuery();
+    const columns = ML.orderableColumns(query);
+
+    it("returns columns from the source and implicitly joinable tables", () => {
+      const expectedCount =
+        getColumnCount(SAMPLE_DATABASE.ORDERS.id) +
+        getColumnCount(SAMPLE_DATABASE.PRODUCTS.id) +
+        getColumnCount(SAMPLE_DATABASE.PEOPLE.id);
+
+      expect(columns).toHaveLength(expectedCount);
+    });
```

Comment:
I don't think these sorts of tests really belong in JS at all. If we wanted to assert that `order-bys` returns some certain number of tables given some metadata, we should do that in the Cljc tests for `order-bys`. 

The only thing we should be testing in JS is stuff we can't test in Cljs, namely that conversion between JS <=> Cljs works automatically and properly

Reply:
For example if we want to tweak this logic in the future to return some additional extra fields, or stop returning some fields from implicitly joinable tables, we should be able to do that and only have to update Clojure(Script) tests , we shouldn't need to update a bunch of FE tests as well, we don't need duplicate coverage for that sort of logic 

The tests that `.find()` stuff sort of go against that idea, but we do need to find something to make sure that the results come back in a usable format at some point, but I think we should try not to do it more than we absolutely have to

Reply:
Yeah, solid point, let me remove them 🧹